sdk/src/lib.rs

```diff
@@ -81,19 +97,30 @@ where
     /// }
     pub async fn build(
         self,
-        indexer_url: Option<String>,
+        indexer_url: Option<&'static str>,
     ) -> Result<(DataStream<F, D>, Sender<Configuration<F>>), ClientBuilderError> {
-        let mut client = self.client.unwrap_or(
-            StreamClient::connect(indexer_url.unwrap_or("http://0.0.0.0:7171".into())).await?,
-        );
+        let channel =
+            Channel::from_static(indexer_url.unwrap_or("https://mainnet.starknet.a5a.ch"))
+                .connect()
+                .await?;
+
+        let mut default_client =
+            StreamClient::with_interceptor(channel, move |mut req: tonic::Request<()>| {
+                if let Some(token) = self.token.clone() {
+                    let token: MetadataValue<_> = format!("Bearer {token}").parse().unwrap();
+                    req.metadata_mut().insert("authorization", token.clone());
```

Comment:
Can remove unnecessary clone.

```suggestion
                    req.metadata_mut().insert("authorization", token);
```